ibtk/include/ibtk/solver_utilities.h

```diff
@@ -0,0 +1,308 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (c) 2021 - 2021 by the IBAMR developers
+// All rights reserved.
+//
+// This file is part of IBAMR.
+//
+// IBAMR is free software and is distributed under the 3-clause BSD
+// license. The full text of the license can be found in the file
+// COPYRIGHT at the top level directory of IBAMR.
+//
+// ---------------------------------------------------------------------
+
+/////////////////////////////// INCLUDE GUARD ////////////////////////////////
+
+#ifndef included_IBTK_solver_utilities
+#define included_IBTK_solver_utilities
+
+/////////////////////////////// INCLUDES /////////////////////////////////////
+
+#include <ibtk/config.h>
+
+#include "petscksp.h"
+#include "petscsnes.h"
+
+IBTK_DISABLE_EXTRA_WARNINGS
+#include "HYPRE_sstruct_ls.h"
+#include "HYPRE_sstruct_mv.h"
+#include "HYPRE_struct_ls.h"
+#include "HYPRE_struct_mv.h"
+IBTK_ENABLE_EXTRA_WARNINGS
+
+#include "CellData.h"
+#include "SideData.h"
+
+#include <array>
+#include <vector>
+
+namespace IBTK
+{
+/*!
+ * \brief Report the KSPConvergedReason.
+ */
+inline void
+reportPETScKSPConvergedReason(const std::string& object_name, const KSPConvergedReason& reason, std::ostream& os)
```

Comment:
These really don't need to be inline. Could you add a source file with the definitions?

Reply:
I've moved all the implementations to a source file.